lib/actions-oclif/tunnel.ts

```diff
@@ -78,7 +77,6 @@ export default class TunnelCmd extends Command {
 		{
 			name: 'deviceOrApplication',
 			description: 'device uuid or application name/id',
-			parse: (x) => tryAsInteger(x),
```

Comment:
This looks like it is going in the right direction, but there is still a corner case. The documentation says that a (numeric) application ID is also accepted:
```
$ balena help tunnel
...
ARGUMENTS
  <deviceOrApplication>  device uuid or application name/id
```
And indeed, the following works where `1301645` is an app ID, not a device UUID:
```
$ balena tunnel 1301645 -p 22222
? Select a device (Use arrow keys)
❯ frosty-snow (dcbff32)
```
This however relies on the code line above that is being deleted:
```
    parse: (x) => tryAsInteger(x),
```
I propose that this line be deleted, but the `tryAsInteger()` call moved to the beginning of this block:
https://github.com/balena-io/balena-cli/blob/v12.14.12/lib/utils/patterns.ts#L392-L395
```
                appOrDevice = tryAsInteger(appOrDevice);
		logger.logDebug(
			`Fetching application by name ${appOrDevice} (${typeof appOrDevice})`,
		);
		const app = await sdk.models.application.get(appOrDevice);
```
Note also how the `getOnlineTargetUuid` function already had several occurrences of `typeof applicationOrDevice` because the code assumed that `applicationOrDevice` might be of type `number` or `string` (and indeed it could and would), despite the argument being declared as a string (which was a type declaration bug).

Note also that `sdk.models.application.get(applicationOrDevice)` behaves differently depending on the type of `applicationOrDevice` being `number` or `string` -- the difference being numeric app ID and an app name, e.g. `1234` _vs._ `'1234'`. For this reason, I think the code should actually be calling `sdk.models.application.get(applicationOrDevice)` twice, once with a numeric argument (if it parses successfully as an integer) and then again with a string argument (if the first call was not successful in retrieving an application).